src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/decisionmaker/actions/ModifyCacheMaxSizeAction.java

```diff
@@ -18,39 +18,54 @@
 import static com.amazon.opendistro.elasticsearch.performanceanalyzer.decisionmaker.actions.ImpactVector.Dimension.HEAP;
 
 import com.amazon.opendistro.elasticsearch.performanceanalyzer.grpc.ResourceEnum;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.configs.CacheDeciderConfig;
 import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.store.rca.cluster.NodeKey;
 import java.util.Collections;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
 
-public class ModifyCacheCapacityAction implements Action {
+public class ModifyCacheMaxSizeAction implements Action {
+    private static final Logger LOG = LogManager.getLogger(ModifyCacheMaxSizeAction.class);
     public static final String NAME = "modifyCacheCapacity";
     public static final long COOL_OFF_PERIOD_IN_MILLIS = 300 * 1_000;
 
-    private long currentCapacityInBytes;
-    private long desiredCapacityInBytes;
+    private long currentCacheMaxSizeInBytes;
+    private long desiredCacheMaxSizeInBytes;
+    private long heapMaxSizeInBytes;
+
+    private double fieldDataCacheSizeUpperBound;
+    private double shardRequestCacheSizeUpperBound;
+
     private ResourceEnum cacheType;
     private NodeKey esNode;
 
     private Map<ResourceEnum, Long> stepSizeInBytes = new HashMap<>();
     private Map<ResourceEnum, Long> upperBoundInBytes = new HashMap<>();
 
-    public ModifyCacheCapacityAction(
+    public ModifyCacheMaxSizeAction(
             final NodeKey esNode,
             final ResourceEnum cacheType,
-            final long currentCapacityInBytes,
+            final long currentCacheMaxSizeInBytes,
+            final long heapMaxSizeInBytes,
+            final double fieldDataCacheSizeUpperBound,
+            final double shardRequestCacheSizeUpperBound,
             final boolean increase) {
-        // TODO: Also consume NodeConfigurationRca
+        this.fieldDataCacheSizeUpperBound = fieldDataCacheSizeUpperBound;
+        this.shardRequestCacheSizeUpperBound = shardRequestCacheSizeUpperBound;
+        this.heapMaxSizeInBytes = heapMaxSizeInBytes;
+
         setBounds();
         setStepSize();
 
         this.esNode = esNode;
         this.cacheType = cacheType;
-        this.currentCapacityInBytes = currentCapacityInBytes;
+        this.currentCacheMaxSizeInBytes = currentCacheMaxSizeInBytes;
         long desiredCapacity =
-                increase ? currentCapacityInBytes + getStepSize(cacheType) : currentCapacityInBytes;
-        setDesiredCapacity(desiredCapacity);
+                increase ? currentCacheMaxSizeInBytes + getStepSize(cacheType) : currentCacheMaxSizeInBytes;
```

Comment:
can we also handle cache scaling down here ?

Reply:
Presently, cache decider only uses increasing the cache max size action. 
I do agree that we need scaling down option as well. I will address them in subsequent PR.

Reply:
Same comment, add a TODO and address that when the JVMDecider is available.

Reply:
Added a TODO